feat/backend/dashboard/student: Changed `Student` properties and Connected UI to Backend

bbcarrots opened the pull request:
I had to refactor the `Student` class to accommodate the integration of UI and backend. Currently, I can forward an update event to the parent class and access the updated values using `ToUpdateStudentStore` however, I need the `studentNumber` and `username` from the UI side to actually update in the database. 

This PR is not yet ready for review. @allainerain I need your help in providing the student number and username for Update, Delete, and Approve.

# Major Changes

## `Student.ts`

- Removed class properties. Replaced it with `toStudentUIObj()` to convert a StudentDBObj to an object the UI can use to display student information.
- Updated `selectStudents()`. `insertStudent()`, `updateStudent()`, `deleteStudent()`, and `approveStudent()` to pass a `StudentDBObj` or the SN and username instead.

## `TableRow.svelte` and all Update-related files
- Added `ToUpdateStudentStore`.
- Added event forwarding from TableRow up to topmost parent page.
- Created `handleCommand()` which creates a PATCH Request.

allainerain commented:
Will it be enough just to pass the student number and not the username? This is because for other tables, there'll only be one primary key, so it would be easier to make the table component adaptable to the other tables if they all just have one primary key.

bbcarrots commented:
The Student Table UI is finally connected with the Backend and Database! Thank you for all the hardwork, @allainerain ! Here are the changes, @gabrielcalubayan72 . Please review! 

# Major Changes
## `supabase.ts`
- Removed `username` as a requirement for CRUD functions.

## `src/routes/api/student/+server.ts`
- Added PATCH and DELETE functions for Approve, Update, and Delete.

## `src/routes/dashboard/students/+page.svelte`
- Added function handlers for Approve, Update, and Delete events for the Student table.

# Minor Changes
## `Table.svelte`
- Renamed event forwarding functions.

## `TableRow.svelte`
- Used one `dispatch` variable for Approve, Delete, and Update.


I have included documentation and formatted the files. Please add your documentation in the `.svelte` files, @allainerain !

allainerain reviewed (approved):
Thank you so much for all the hard work @bbcarrots! I'll work on adding documentation to the components in the frontend :D
